### PR TITLE
Add castable-instances for tuple and gels-function

### PR DESCRIPTION
**Pull request by junjihashimoto**

### CODE REVIEW

File: hasktorch/src/Torch/Tensor.hs [hunk 1]
```diff
@@ -202,6 +205,82 @@ instance TensorLike [[[Double]]] where
...
+--------------------------------------------------------------------------------
+
+instance Castable (Tensor,Tensor) (ForeignPtr (ATen.Tensor,ATen.Tensor)) where
+  cast _ _ = undefined
+  uncast t f = do
+    t0 <- get0 t
+    t1 <- get1 t
+    f (Unsafe t0, Unsafe t1)
+
+instance Castable (Tensor,Tensor,Tensor) (ForeignPtr (ATen.Tensor,ATen.Tensor,ATen.Tensor)) where
```
Comment (apaszke) [inline hasktorch/src/Torch/Tensor.hs]: Can't we generate instances like `(Castable a a', Castable b b', Castable c c') => Castable (a, b, c) (a', b', c')` as part of the ATen codegen?

Comment (junjihashimoto) [reply hasktorch/src/Torch/Tensor.hs]: Yes, we can. I think so.
In addition to this, we can generate simple funtions like gels, though we have to write the test.

Comment (junjihashimoto) [reply hasktorch/src/Torch/Tensor.hs]: For now, I am thinking how to implement `(Castable a a', Castable b b', Castable c c') => Castable (a, b, c) (a', b', c')`.

Comment (junjihashimoto) [reply hasktorch/src/Torch/Tensor.hs]: The implementation is done.

**Pull request merged**